geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/utils/HBaseVersions.scala

```diff
@@ -8,62 +8,106 @@
+      bloom.foreach(_setBloomFilterType(column, _))
+      compression.foreach(_setCompressionType(column, _))
+      encoding.foreach(_setDataBlockEncoding(column, _))
+      inMemory.foreach(_setInMemory(column, _))
+      _addFamily(descriptor, column)
     }
-    val parameterTypes = method.getParameterTypes
-    if (parameterTypes.length == 1 && parameterTypes.head == classOf[HColumnDescriptor]) {
-      (descriptor, family) => method.invoke(descriptor, family)
-    } else {
-      throw new NoSuchMethodException(s"Couldn't find HTableDescriptor.addFamily method with correct parameters: $method")
+
+    coprocessor.foreach { case (clas, path) =>
+      _addCoprocessor(descriptor, clas, path.orNull, Coprocessor.PRIORITY_USER, null)
     }
+
+    _createTableAsync(admin, descriptor, if (splits.isEmpty) { null } else { splits.toArray })
   }
 
-  private val _addCoprocessor: (HTableDescriptor, String, Path, Int, java.util.Map[String, String]) => Unit = {
-    val method = hTableDescriptorMethods.find(m => m.getName == "addCoprocessor" && m.getParameterCount == 4).getOrElse {
+  /**
+   * Disable a table asynchronously
+   *
+   * @param admin admin hbase connection
+   * @param table table to disable
+   */
+  def disableTableAsync(admin: Admin, table: TableName): Unit = _disableTableAsync(admin, table)
+
+  /**
+   * Checks whether HBase is available, and throws an exception if not
+   *
+   * @param conf HBase configuration
+   */
+  def checkAvailable(conf: Configuration): Unit = _available(conf)
+
+  private lazy val hTableDescriptorClass = Class.forName("org.apache.hadoop.hbase.HTableDescriptor")
+
+  private lazy val hColumnDescriptorClass = Class.forName("org.apache.hadoop.hbase.HColumnDescriptor")
+
+  private lazy val _setBloomFilterType: (Any, BloomType) => Unit =
+    findMethod(hColumnDescriptorClass, "setBloomFilterType", classOf[BloomType])
+
+  private lazy val _setCompressionType: (Any, Algorithm) => Unit =
+    findMethod(hColumnDescriptorClass, "setCompressionType", classOf[Algorithm])
+
+  private lazy val _setDataBlockEncoding: (Any, DataBlockEncoding) => Unit =
+    findMethod(hColumnDescriptorClass, "setDataBlockEncoding", classOf[DataBlockEncoding])
+
+  private lazy val _setInMemory: (Any, Boolean) => Unit =
+    findMethod(hColumnDescriptorClass, "setInMemory", classOf[Boolean])
+
+  // HBase 1.3 signature: public HTableDescriptor addFamily(final HColumnDescriptor family)
+  // CDH 5.12 signature: public void addFamily(final HColumnDescriptor family)
+  private lazy val _addFamily: (Any, Any) => Unit =
+    findMethod(hTableDescriptorClass, "addFamily", hColumnDescriptorClass)
+
+  private lazy val _disableTableAsync: (Admin, TableName) => Unit =
+    findMethod(classOf[Admin], "disableTableAsync", classOf[TableName])
+
+  // HBase 1.3 signature:
+  // public HTableDescriptor addCoprocessor(String className, Path jarFilePath, int priority, final Map&lt;String, String&gt; kvs)
+  // CDH 5.12 signature:
```

Comment:
What version of HBase does CDH 5.12 ship?!

Reply:
uhh, that was a leftover comment from some time ago... i forget now